File: src/teste-digi-v2/app/Http/Controllers/ProductController.php
<?php

namespace App\Http\Controllers;

use App\Repository\ProductRepository;
use Illuminate\Http\Request;

class ProductController extends Controller
{
    /**
     * @OA\Get(
     *      path="/api/product/getAll",
     *      tags={"Product"},
     *      summary="Lista de produtos",
     *      description="Retorna a lista de produtos oriunda do arquivo JSON",
     *      @OA\Response(
     *          response=200,
     *          description="Solicitacão efetuada com sucesso"
     *      ),
     *      @OA\Response(
     *          response=500,
     *          description="Erro interno do servidor"
     *      )
     *    )
     * 
     * Retorna a lista de produtos oriunda do Json
     */
    public function getAll()
    {
        $repository = new ProductRepository();

        $products = $repository->getProducts();

        return response()->json($products);
    }
}

File: src/TesteDigi/TesteDigi.Api/Controllers/ProductController.cs
using Microsoft.AspNetCore.Mvc;
using System.Net;
using TesteDigi.Repository;
using TesteDigi.Service;

namespace TesteDigi.Api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ProductController : ControllerBase
    {
        private readonly ProductService _productService;

        public ProductController(ProductService productService)
        {
            _productService = productService;
        }

        /// <summary>
        /// Listar Produtos
        /// </summary>
        /// <response code="500">Internal Server Error</response>
        /// <remarks>Obtém a lista de produtos oriunda do arquivo JSON</remarks>
        /// <returns></returns>
        [HttpGet("getAll")]
        public IActionResult GetAll()
        {
            try
            {
            var o = _productService.GetAll();

            return Ok(o);

            }
            catch(Exception ex)
            {
                return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
            }
        }
    }
}
